src/parser.ts

```diff
@@ -3266,9 +3245,13 @@ export class Parser extends DiagnosticEmitter {
       ) {
         let args = this.parseArguments(tn);
         if (!args) return null;
-        expr = Node.createCallExpression(expr, typeArguments, args, tn.range(startPos, tn.pos)); // is again callable
+        expr = (isNew ? Node.createNewExpression : Node.createCallExpression)(
```

Comment:
This'll compile to an indirect call with two table elements in WASM. If possible, we should probably avoid that.